slapo/model_schedule/registry.py

```diff
@@ -0,0 +1,28 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Model schedule registration."""
+import inspect
+
+# Mapping from model name to schedule method.
+SCHEDULE_METHODS = {}
+
+
+def get_schedule_method(model_name):
+    """Get the schedule method."""
+    if model_name not in SCHEDULE_METHODS:
+        return None
+    return SCHEDULE_METHODS[model_name]
+
+
+def register_schedule_method():
```

Comment:
Since this could also be a user API (in the case they want to register a custom schedule), I would suggest a concise name "register_schedule".